core/src/main/scala/com/salesforce/op/stages/impl/selector/SelectedCombiner.scala

```diff
@@ -0,0 +1,245 @@
+) extends Estimator[SelectedCombinerModel] with
+  OpPipelineStage3[RealNN, Prediction, Prediction, Prediction] with SelectedCombinerParams with HasEval {
+
+  override def evaluators: Seq[OpEvaluatorBase[_ <: EvaluationMetrics]] = {
+    val ev1 = in2.getFeature().originStage.asInstanceOf[ModelSelector[_, _]].evaluators
+    val ev1names = ev1.map(_.name).toSet
+    val ev2 = in2.getFeature().originStage.asInstanceOf[ModelSelector[_, _]].evaluators
+    ev1 ++ ev2.filter(e => ev1names.contains(e.name))
+  }
+
+  override protected[op] def outputsColNamesMap: Map[String, String] =
+    Map(ModelSelectorNames.outputParamName -> getOutputFeatureName)
+
+  override protected[op] def labelColName: String = in1.name
+
+  private def getSummary(feature: TransientFeature) =
+    ModelSelectorSummary.fromMetadata(getInputSchema()(feature.name).metadata.getSummaryMetadata())
+
+  override def onSetInput(): Unit = {
+    super.onSetInput()
+
+    require(
+      in2.getFeature().originStage.isInstanceOf[ModelSelector[_, _]] &&
+        in3.getFeature().originStage.isInstanceOf[ModelSelector[_, _]],
+      "Predictions must be from model selectors - other types of model are not supported at this time"
+    )
+
+  }
+
+  override def fit(dataset: Dataset[_]): SelectedCombinerModel = {
+    setInputSchema(dataset.schema).transformSchema(dataset.schema)
+
+    val summary1 = getSummary(in2)
+    val summary2 = getSummary(in3)
+
+    require(summary1.problemType == summary2.problemType,
+      s"Cannot combine model selectors for different problem types found ${summary1.problemType}" +
+        s" and ${summary2.problemType}")
+
+    def getMetricValue(metrics: EvaluationMetrics, name: EvalMetric) = metrics.toMap.get(name.humanFriendlyName)
+      .map(_.asInstanceOf[Double])
+
+    def getWinningModelMetric(summary: ModelSelectorSummary) = {
+      summary.validationResults.collectFirst {
+        case r if r.modelUID == summary.bestModelUID =>
+          getMetricValue(r.metricValues, summary.evaluationMetric)
+      }.flatten
+    }
+
+    def getMet(optionMet: Option[Double]) = optionMet.getOrElse {
+      throw new RuntimeException("Evaluation metrics for two model selectors are non-overlapping")
+    }
+
+
+    val eval1 = summary1.evaluationMetric
+    val eval2 = summary2.evaluationMetric
+
+    val (metricValueOpt1: Option[Double], metricValueOpt2: Option[Double], metricName: EvalMetric) =
+      if (eval1 == eval2) { // same decision metric in validation results
+        (getWinningModelMetric(summary1), getWinningModelMetric(summary2), eval1)
+      } else { // look for overlapping metrics in training results
+        val m2e1 = getMetricValue(summary2.trainEvaluation, eval1)
+        val m1e2 = getMetricValue(summary1.trainEvaluation, eval2)
+        if (m2e1.nonEmpty) {
+          (getMetricValue(summary1.trainEvaluation, eval1), m2e1, eval1)
+        } else if (m1e2.nonEmpty) {
+          (m1e2, getMetricValue(summary2.trainEvaluation, eval2))
```

Comment:
Don't you need an `eval2` at the end of the tuple here? How did the Scala compiler miss that actually??

Can you add some tests that cover some of these weird edge cases? I feel like there's a lot of new stuff being added.